packages/gatsby-plugin-utils/src/polyfill-remote-file/graphql/gatsby-image-resolver.ts

```diff
@@ -169,31 +168,29 @@ export async function gatsbyImageResolver(
         dispatchLocalImageServiceJob(
           {
             url: source.url,
-            extension: format,
-            basename: path.basename(
-              source.filename,
-              path.extname(source.filename)
-            ),
+            mimeType: source.mimeType,
+            filename: source.filename,
+            contentDigest: source.internal.contentDigest,
+          },
+          {
             width,
             height: Math.round(width / imageSizes.aspectRatio),
             format,
-            fit: args.fit as ImageFit,
-            contentDigest: source.internal.contentDigest,
+            // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+            cropFocus: args.cropFocus!,
```

Comment:
Why could this be `undefined`?

Reply:
I think we had bad types before, thanks!